rust-src/id/src/id_verifier.rs

```diff
@@ -0,0 +1,173 @@
+    hasher.update(b"account_ownership_proof");
+    hasher.update(&challenge);
+    let to_sign = &hasher.finalize();
+    utils::verify_account_ownership_proof(&public_data.keys, public_data.threshold, &proof, to_sign)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{constants::AttributeKind, id_prover::*};
+    use crypto_common::types::{KeyIndex, KeyPair};
+    use pairing::bls12_381::G1;
+    use rand::*;
+    use std::collections::btree_map::BTreeMap;
+
+    #[test]
+    fn test_verify_account_ownership() {
+        let mut csprng = thread_rng();
+
+        let cred_data = CredentialData {
+            keys:      {
+                let mut keys = BTreeMap::new();
+                keys.insert(KeyIndex(0), KeyPair::generate(&mut csprng));
+                keys.insert(KeyIndex(1), KeyPair::generate(&mut csprng));
+                keys.insert(KeyIndex(2), KeyPair::generate(&mut csprng));
+                keys
+            },
+            threshold: SignatureThreshold(2),
+        };
+
+        let pub_data = cred_data.get_cred_key_info();
+
+        let reg_id: G1 = Curve::hash_to_group(b"some_bytes");
+        let account_address = AccountAddress::new(&reg_id);
+        let challenge = b"13549686546546546854651357687354";
+
+        let proof = prove_ownership_of_account(cred_data, account_address, challenge);
+
+        assert!(verify_account_ownership(
+            &pub_data,
+            account_address,
+            challenge,
+            &proof
+        ));
+    }
+
+    #[test]
+    fn test_verify_attribute() {
+        let mut csprng = thread_rng();
+        let keys = PedersenKey::<G1>::generate(&mut csprng);
+        let attribute = AttributeKind("some attribute value".to_string());
+        let value = Value::<G1>::new(attribute.to_field_element());
+        let (commitment, randomness) = keys.commit(&value, &mut csprng);
+
+        assert!(verify_attribute(
+            &keys,
+            &attribute,
+            &randomness,
+            &commitment
+        ));
+    }
+
+    #[test]
+    fn test_verify_attribute_in_range() {
+        let mut csprng = thread_rng();
+        let global = GlobalContext::<G1>::generate(String::from("genesis_string"));
+        let keys = global.on_chain_commitment_key;
+        let gens = global.bulletproof_generators();
+        let lower = AttributeKind("20000102".to_string());
+        let attribute = AttributeKind("20000102".to_string());
+        let upper = AttributeKind("20000103".to_string());
+        let value = Value::<G1>::new(attribute.to_field_element());
+        let (commitment, randomness) = keys.commit(&value, &mut csprng);
+        let maybe_proof =
+            prove_attribute_in_range(&gens, &keys, &attribute, &lower, &upper, &randomness);
+        let result = if let Some(proof) = maybe_proof {
+            verify_attribute_range(&keys, &gens, &lower, &upper, &commitment, &proof)
+        } else {
+            false
+        };
+        assert!(result);
```

Comment:
Recommend adding a description of what the assertion is doing so that if it fails it is possible to find what went wrong.